web/src/engine/osk/src/views/floatingOskView.ts

```diff
@@ -178,55 +168,51 @@ export default class FloatingOSKView extends OSKView {
   /**
    * Save size, position, font size and visibility of OSK
    */
-  saveCookie() {
+  private savePersistedLayout() {
     var p = this.getPos();
 
     const c: FloatingOSKCookie = {
-      visible: this.displayIfActive ? '1' : '0',
-      userSet: this.userPositioned ? '1' : '0',
-      left: '' + p.left,
-      top: '' + p.top,
+      visible: this.displayIfActive ? 1 : 0,
+      userSet: this.userPositioned ?  1 : 0,
+      left: p.left,
+      top:  p.top,
       _version: Version.CURRENT.toString()
     }
 
     if(this.vkbd) {
-      c['width'] = '' + this.width.val;
-      c['height'] = '' + this.height.val;
+      c.width =  this.width.val;
+      c.height = this.height.val;
     }
 
-    const typedConfiguration = this.configuration as FloatingOSKViewConfiguration;
-    typedConfiguration.saveViewLayout(c);
+    this.layoutSerializer.save(c as Required<FloatingOSKCookie>);
   }
 
   /**
    * Restore size, position, font size and visibility of desktop OSK
    *
    *  @return {boolean}
    */
-  loadCookie(): void {
-    function parseIntWithDefault(str: string, fallback: number) {
-      let val = Number.parseInt(str, 10);
-      return isNaN(val) ? fallback: val;
-    }
-
-    const typedConfiguration = this.configuration as FloatingOSKViewConfiguration;
-    var c: FloatingOSKCookie = typedConfiguration.reloadViewLayout();
+  private loadPersistedLayout(): void {
+    let c = this.layoutSerializer.loadWithDefaults({
+      visible: 1,
+      userSet: 0,
+      left: -1,
+      top: -1,
+      _version: undefined,
+      width:  0.3*screen.width,
+      height: 0.15*screen.height
+    });
 
-    this.activationModel.enabled = parseIntWithDefault(c['visible'], 1) == 1;
-    this.userPositioned = parseIntWithDefault(c['userSet'], 0) == 1;
-    this.x = parseIntWithDefault(c['left'],-1);
-    this.y = parseIntWithDefault(c['top'],-1);
-    let cookieVersionString = c['_version'];
+    this.activationModel.enabled = c.visible == 1;
+    this.userPositioned = c.userSet == 1;
+    this.x = c.left;
+    this.y = c.top;
+    const cookieVersionString = c._version;
 
     // Restore OSK size - font size now fixed in relation to OSK height, unless overridden (in em) by keyboard
-    let dfltWidth=0.3*screen.width;
-    let dfltHeight=0.15*screen.height;
-
-    let newWidth  = parseInt(c['width'], 10);
-    let newHeight = parseInt(c['height'], 10);
-    let isNewCookie = isNaN(newHeight);
-    newWidth  = isNaN(newWidth)  ? dfltWidth  : newWidth;
-    newHeight = isNaN(newHeight) ? dfltHeight : newHeight;
+    const isNewCookie = cookieVersionString === undefined;
+    let newWidth  = c.width;
+    let newHeight = c.height;
```

Comment:
How will this work with existing cookies?

Reply:
https://github.com/keymanapp/keyman/blob/a8bd8a1712bac282e160cd3e9027c68905a78083/web/src/engine/osk/src/views/floatingOskView.ts#L195-L204

If the original cookie does not provide values for these fields, the default values set by the block above are used instead.  It should exhibit the same behavior as the original cookie implementation.  It's like the old `parseIntWithDefault`, except the defaults are provided within a full, completely-specified object for all cookie-object properties at once, rather than one at a time.  The type-specification setup will ensure that no properties are omitted within the default object.  (For example, TS will generate a compile-time error if a new cookie field is added without a corresponding default being defined... or even if the value is of the wrong type, based on the typing of the cookie's `interface` spec.)

I debated a bit on whether or not to make the `loadWithDefaults` method a base-class method... and I'm still considering the possibility.